### PR TITLE
Fix demo program provisioning on BSD/Macs

### PR DESCRIPTION
**Pull request by mikix**
And some minor other fixes, commented below.

### CODE REVIEW

File: course-generator/build-course-json.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Script to build course configurations in proper json format that can be passed into provision-courses.sh
+# Script to build course configurations in proper json format that can be passed into create-courses.sh
```
Comment (mikix) [inline course-generator/build-course-json.sh:1]: Minor typo I saw when doing some other work today. It's referencing the script in the same folder by the wrong name.

File: programs/provision.sh [hunk 2]
```diff
@@ -28,7 +28,7 @@ notice() {
 reset_color() {
     SWITCH="\033["
     NORMAL="${SWITCH}0m"
-    echo "${NORMAL}"
+    echo -n "${NORMAL}"
```
Comment (mikix) [inline programs/provision.sh:31]: Don't add an extra newline in our output when going back to the normal color.

File: programs/provision.sh [hunk 3]
```diff
@@ -50,7 +50,7 @@ provision_ida() {
     cmd=${2:-"shell"}
 
     # Escape double quotes and backticks from the Python
-    PROGRAM_SCRIPT="$(sed 's/\("\|`\)/\\\1/g' < "$BASEDIR/$service.py")"
+    PROGRAM_SCRIPT="$(sed -E 's/("|`)/\\\1/g' < "$BASEDIR/$service.py")"
```
Comment (mikix) [inline programs/provision.sh:53]: Uses a regex format that is standard across POSIX, rather than the default format, which apparently can differ.

**Review by bderusha — APPROVED** [review]
🎸

**Pull request merged**